components/CompanyRegister.tsx

```diff
@@ -23,14 +26,49 @@ const CompanyRegister = () => {
     setCompanySite(event.target.value);
   };
 
+  db.signer(async (data: string) => {
+    const sig = await eth.sign(data, address);
+    return { h: "eth-personal-sign", sig };
+  });
+
+  const { data, isLoading, isSuccess, writeAsync } = useContractWrite({
+    abi: recruitmentABI,
+    address: recruitmentAddress,
+    functionName: "registerCompany",
+  });
+
+  const registerCompanySc = async () => {
+    try {
+      const companyData = [
+        address,
+        companyName,
+        companyDescription,
+        companySite,
+      ];
+      await writeAsync();
+      if (!isSuccess) {
+        toast.error("Company Register failed");
+      }
+      if (isSuccess && !isLoading) {
+        const company = await registerCompany(companyData);
```

Comment:
If polybase base write fails - then we have a floating company in the SC.

These should be transactional...

For momentum happy for a ticket to be created to come back and fix this.

Reply:
sure! pls add the ticket

Reply:
Tracking ticket added:

https://www.notion.so/Data-syncing-between-polybase-and-SC-needs-to-be-transactional-90237a130f134b53bda65623e5937cbf?pvs=4